### PR TITLE
Overriden how transaction receipt is marshalled to json.

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why is this change needed?

Fix for #970 

### What changes were made as part of this PR:

Changed how receipt is being marshalled. Using a cloned function instead of the standard one in the geth library which is not compliant with the library. **It is a bit of a hack, which makes me uneasy, but there is no other way to encode null**

### Definition of done

- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated 
- [ ] End-to-end tests run if required (see below)

### End-to-end tests

Running the end-to-end tests in the [obscuro-test repo](https://github.com/obscuronet/obscuro-test) is currently at the 
discretion of the developer prior to merging a PR. The intention is not to slow down development by having the longer 
running end-to-end tests as a PR gate run on each branch commit etc. To manually trigger a run pre-merge;

- Go to [run_local_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_local_tests.yml)
- Click the "Run workflow" button
- Use the main branch of obscuro-test 
- Enter the name of the PR branch for go-obscuro
- Run the workflow

The run takes approximately 20 minutes and any failure will be reported in the workflow output. Should the tests fail 
the docker container output for the local testnet and all test artifacts will be added to the run and can be downloaded
for debugging and analysis, with a retention period of 2 days. 

Note that every PR merge to main will also trigger a run of post-merge tests, so even if you do not manually trigger 
pre-merge, tests will be run post-merge. The intention being to catch any issues fast on main to allow for quick 
resolution. The post-merge test output can be seen at 
[run_merge_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_merge_tests.yml) and will show both 
the PR number and author in the list of runs.

### CODE REVIEW

**Comment by StefanIliev545** [comment]
Alternatively we can use something that converts the struct to a map and removes the contract address if it matches zero address, however this still overrides how json is generated and I feel it makes it more obfuscated.

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -510,11 +510,13 @@ func (e *enclaveImpl) GetTransactionReceipt(encryptedParams common.EncryptedPara
 	}
 
 	// We marshal the receipt to JSON.
-	txReceiptBytes, err := txReceipt.MarshalJSON()
+	txReceiptBytes, err := common.MarshalReceipt(txReceipt)
 	if err != nil {
 		return nil, fmt.Errorf("could not marshal transaction receipt to JSON in eth_getTransactionReceipt request. Cause: %w", err)
 	}
 
+	fmt.Println(string(txReceiptBytes))
```
Comment (otherview) [inline go/enclave/enclave.go:518]: Lingering print statement.

```suggestion
```

File: go/common/json_types.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	enc.Status = hexutil.Uint64(r.Status)
+	enc.CumulativeGasUsed = hexutil.Uint64(r.CumulativeGasUsed)
+	enc.Bloom = r.Bloom
+	enc.Logs = r.Logs
+	enc.TxHash = r.TxHash
+	// Contract address should only be set for receipts of transactions that initialize a contract.
+	// Otherwise return nil to match receipts returned by other testnets.
+	if !bytes.Equal(r.ContractAddress.Bytes(), gethcommon.BigToAddress(gethcommon.Big0).Bytes()) {
+		enc.ContractAddress = &r.ContractAddress
+	}
```
Comment (otherview) [inline go/common/json_types.go:43]: geth to always return `0x0` contract address when contract address is nil. Is there a reason to change this condition ?

**Review by otherview — COMMENTED** [review]
I think our guideline is to replicate geth. I wouldn't necessarily change this.. unless we're missing something.

I've updated https://github.com/obscuronet/obscuro-internal/issues/970 with the comment.
It might be that we're drawing incorrect conclusions of bug. ie. We need to replicate the issue described in the test in a geth network, ropstein is run by infura which has a whole myriad of servers running.

**Pull request closed without merging**